examples/guestbook-go/redis-master-service.json

```diff
@@ -12,7 +12,8 @@
       "ports": [
         {
           "port":6379,
-          "targetPort":"redis-server"
+          "targetPort":"redis-server",
+          "nodePort":0
```

Comment:
You're right. In [types.go](https://github.com/kubernetes/kubernetes/blob/master/pkg/api/v1/types.go#L1181) `nodePort` is marked as `json:"nodePort,omitempty"` and should be optional. It's weird that [swagger](http://kubernetes.io/third_party/swagger-ui/#!/api%2Fv1/createNamespacedService) didn't mark it as optional:
```yaml
v1.ServicePort {
  name (string, optional): the name of this port; optional if only one port is defined ,
  protocol (string, optional): the protocol used by this port; must be UDP or TCP; TCP if unspecified ,
  port (integer): the port number that is exposed ,
  targetPort (string, optional): number or name of the port to access on the pods targeted by the service; defaults to the service port; number must be in the range 1 to 65535; name must be an IANA_SVC_NAME; see http://releases.k8s.io/HEAD/docs/services.md#defining-a-service ,
  nodePort (integer): the port on each node on which this service is exposed when type=NodePort or LoadBalancer; usually assigned by the system; if specified, it will be allocated to the service if unused or else creation of the service will fail; see http://releases.k8s.io/HEAD/docs/services.md#type--nodeport
}
```